### DISCUSSION TITLE
Table to count number of tasks per language

### DISCUSSION THREAD

**Discussion by isaac-chung** [embeddings-benchmark/mteb#686] [General]
With growing numbers of datasets in multilingual part of MTEB, maybe a table is needed to tally how many datasets there are per task per language. I would imagine something like this is also needed for the submission (e.g. some stats grouped by language family). Where would this live? Maybe as a script under docs/mmteb? Or a separate repo like https://github.com/embeddings-benchmark/mtebscripts? or adding to that repo?

Tagging @imenelydiaker @KennethEnevoldsen @Muennighoff and anyone who might be interested.

For example:
| Language | BitextMining | Classification | Clustering | InstructionRetrieval | PairClassification | Reranking | Retrieval | STS | Summarization |
|---|------|------|------|------|------|------|------|------|---|
| ace | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |
| acm | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |

**Comment by KennethEnevoldsen** [comment]
We can add it at the top of the existing task overview. You can see how the previous script inserts the table at a specific spot.

Unsure if we should do two tables one for the group languages and one for language groups or do a nested table:

| Language | BitextMining | Classification | Clustering | InstructionRetrieval | PairClassification | Reranking | Retrieval | STS | Summarization |
|---|------|------|------|------|------|------|------|------|---|
| **Indo-European** | 0 | 2 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |
| dan | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |
| nob | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |

**Reply by isaac-chung** [reply]
Sure. Let's append at the end of the `tasks.md` file. As for grouping, we could group by genus (e.g. Indic, Romance) or Family (e.g. Indo-European, Niger-Congo).
I don't think cell merging is supported in Github flavoured markdown. Maybe we keep separate tables for groups and all languages?

**Reply by KennethEnevoldsen** [reply]
Feel free to do it for whatever is the easiest to get the metadata for.

**Comment by KennethEnevoldsen** [comment]
For the paper, we might also create an overview similar to:

<img width="868" alt="Screenshot 2024-05-14 at 09 21 35" src="https://github.com/embeddings-benchmark/mteb/assets/23721977/5e187b99-e9da-48be-918d-7da988e07562">

Also seen e.g. [here](https://openreview.net/pdf/f5f1953a9c798ec61bb050e62bc7a94037fd4fab.pdf)

We probably don't want to fields to be datasets but maybe rather domains e.g. something like: 
<img width="797" alt="Screenshot 2024-05-14 at 09 43 10" src="https://github.com/embeddings-benchmark/mteb/assets/23721977/632db1c3-2618-4b1c-be94-006ade479ba5">

or based on subtasks:
<img width="899" alt="Screenshot 2024-05-14 at 09 41 44" src="https://github.com/embeddings-benchmark/mteb/assets/23721977/f5579a28-ce22-47a0-9e05-81a88eb3c325">

**Reply by imenelydiaker** [reply]
I'd go for the latest scratch, easier to read and less dense

**Reply by KennethEnevoldsen** [reply]
I have added a figure to overleaf

**Reply by Muennighoff** [reply]
Looks great! I think we need to be a bit more clear about what we mean by `tasks`. What was `datasets` in the original  MTEB figure seems to have become `tasks`. If we all prefer to refer to these as tasks, I think that's fine but then maybe what was `tasks` in the original MTEB figure also needs a new name, say `abstract tasks` or `task types`. What do you think?

It's also a bit confusing in some parts of the codebase, e.g. in https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md we alternate between calling them datasets & tasks.

Also curious what @orionw thinks is the best way :)

**Reply by orionw** [reply]
The word `task` is definitely overloaded :) 

I think we should use `dataset` to refer to an individual dataset under a specific `task` (or `task type` if we prefer, I like it - helps distinguish it from being conflated with `dataset`). 

The [FLAN paper](https://arxiv.org/pdf/2210.11416) called these `task categories`, maybe these definitions can help?  I think these definitions differ according to each paper so I'm not attached to their definitions, but it would benefit us to be consistent on these definitions.

![image](https://github.com/embeddings-benchmark/mteb/assets/31665361/87b22694-8f0a-4caa-91a9-63d44df57dd3)

**Reply by imenelydiaker** [reply]
@orionw The FLAN notation seems great, it looks exactly like what we're trying to do. Let's go for this:

**Dataset** -> e.g., MLSUM, AmazonReviews
**Task Category** -> Classification, Clustering, Retrieval, etc.
**Task** -> Dataset x Task Category : e.g., MLSUMClustering, AmazonReviewsClassification

**Comment by imenelydiaker** [comment]
Should we make `mmtebscripts` public for storing the scripts that we'll use to run experiments?

**Reply by KennethEnevoldsen** [reply]
Why not script keep it in this repo?

**Reply by imenelydiaker** [reply]
I just thought it would be better for reproducing the paper's results, but no strong preferences

**Reply by KennethEnevoldsen** [reply]
I think I would personally prefer having it all in one repo

**Comment by dokato** [comment]
I think it's a good idea. To facilitate this just want to bring to your attention @isaac-chung a (fairly) new method that should make it easy to aggregate the data this way, e.g.:

```py
task_types = ["BitextMining","Classification","Clustering","InstructionRetrieval","PairClassification","Reranking","Retrieval","STS","Summarization"]
for tt in task_types:
    print(get_tasks(task_types=[tt]).count_languages())
```

**Reply by isaac-chung** [reply]
Thanks! I'll take a look.
